### PR TITLE
feat: allow to specify folder for init

### LINKED ISSUE #556: Add `iotedgedev init` an option to initalize in a new folder
### Description 

As a developer I'd to be able to specify a folder iotedgedev is initialized in, instead of having it initialized in the current directory.

### Additional notes

Possible implementation would be to add an optional `--folder` flag to `iotedgedev init`:

```sh
iotedgedev init --folder my_new_iot_edge_folder
```

### PR DESCRIPTION
**Pull request by roel4ez**
## Checklist

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] Versions update reflected in all places (both __init__.py files, CHANGELOG, setup.py, setup.cfg)
- [x] Unit tests pass locally
- [x] Quickstart steps locally validated
- [ ] Passes unit tests in CICD pipeline (green on Github pipeline)
- [ ] Pypi RC version passes Edge CICD pipeline validation for cross-tool validation
- [x] Pull request includes test coverage for the included changes
- [x] Documentation is updated for the included changes

### General guidelines

- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.

## Description

Fixes #556

Adds an argument to the `iotedgedev init` command, which specifies the name of the folder where the solution will be created. Defaults to `.` if not provided.

### Additional information

Changed the way the `init` command is calling is calling the `create` command: it now calls the constructor of the Solution directly instead of going through the `call_proc`.

### CODE REVIEW

File: iotedgedev/cli.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+from email.policy import default
```
Comment (marianan) [inline iotedgedev/cli.py:1]: what is this library needed for?

Comment (roel4ez) [reply iotedgedev/cli.py:1]: sorry I missed that, no idea how that came in. It's not needed

Comment (roel4ez) [reply iotedgedev/cli.py]: I blame GitHub co-pilot 😄 
But, I should have spotted the change before committing it - sorry about that.

**Review by marianan — APPROVED** [review]
:shipit:

**Comment by roel4ez** [comment]
sorry this was taking a while - has a busy week. Looking into it again.

File: tests/test_iotedgedev_solution.py [hunk 2]
```diff
@@ -110,6 +111,30 @@ def test_solution_create_in_empty_child_dir(prepare_solution_with_env):
     assert 'AZURE IOT EDGE SOLUTION CREATED' in result.output
 
 
+def test_solution_init_without_name():
```
Comment (EliiseS) [inline tests/test_iotedgedev_solution.py:114]: Could we move these test into it's own file under `test_iotedgedev_solution_init.py`?

**Pull request merged**